src/core/lib/attribute-transition-manager.js

```diff
@@ -0,0 +1,341 @@
+    }
+
+    if (needsRedraw) {
+      this._runTransformFeedback({uniforms, buffers});
+    }
+
+    return needsRedraw;
+  }
+  /* eslint-enable max-statements */
+
+  /* Private methods */
+  // Redraw the transform feedback
+  _runTransformFeedback({uniforms, buffers}) {
+    const {model, transformFeedback} = this;
+
+    // Cannot bind to transform feedback if it's already used as attribute
+    Object.values(buffers).forEach(buffer => buffer.unbind());
+
+    model.program.use();
+    transformFeedback.bindBuffers(buffers, {clear: true});
+    transformFeedback.begin(GL.POINTS);
+
+    const parameters = {
+      [GL.RASTERIZER_DISCARD]: true
+    };
+
+    model.draw({uniforms, parameters});
+
+    transformFeedback.end();
+  }
+
+  // Create a model for the transform feedback
+  _createModel() {
+    // Build shaders
+    const varyings = [];
+    const attributeDeclarations = [];
+    const uniformsDeclarations = [];
+    const varyingDeclarations = [];
+    const calculations = [];
+
+    for (const attributeName in this.attributeTransitions) {
+      const transition = this.attributeTransitions[attributeName];
+      const attributeType = ATTRIBUTE_MAPPING[transition.attribute.size];
+
+      if (attributeType) {
+        transition.bufferIndex = varyings.length;
+        varyings.push(attributeName);
+
+        attributeDeclarations.push(`attribute ${attributeType} ${attributeName}From;`);
+        attributeDeclarations.push(`attribute ${attributeType} ${attributeName}To;`);
+        uniformsDeclarations.push(`uniform float ${attributeName}Time;`);
+        varyingDeclarations.push(`varying ${attributeType} ${attributeName};`);
+        calculations.push(`${attributeName} = mix(${attributeName}From, ${attributeName}To,
+          ${attributeName}Time);`);
+      }
+    }
+
+    const vs = `
+#define SHADER_NAME feedback-vertex-shader
+${attributeDeclarations.join('\n')}
+${uniformsDeclarations.join('\n')}
+${varyingDeclarations.join('\n')}
+
+void main(void) {
+  ${calculations.join('\n')}
+  gl_Position = vec4(0.0);
+}
+`;
+
+    const fs = `\
+#define SHADER_NAME feedback-fragment-shader
+
+#ifdef GL_ES
+precision highp float;
+#endif
+
+${varyingDeclarations.join('\n')}
+
+void main(void) {
+  gl_FragColor = vec4(0.0);
+}
+`;
+
+    if (this.model) {
+      this.model.destroy();
+    }
+
+    this.model = new Model(this.gl, {
+      id: this.id,
+      program: new Program(this.gl, {vs, fs, varyings}),
+      geometry: new Geometry({
+        id: this.id,
+        drawMode: GL.POINTS
+      }),
+      vertexCount: 0,
+      isIndexed: true
+    });
+
+    this._updateModel(this.attributeTransitions);
+  }
+
+  // get current values of an attribute, clipped/padded to the size of the new buffer
+  _getCurrentAttributeState(transition) {
+    const {attribute, buffer} = transition;
+    const {value, type, size} = attribute;
+
+    if (buffer) {
+      // No entrance transition
+      let oldBufferData = new Float32Array(value);
```

Comment:
Lines 248 to 252, can be replaced with just one line:
`oldBufferData = buffer.getData()`
`buffer.getData()` should allocate correct type and sized array , copy contents and return.

Reply:
New attribute value may be a different length. We need to convert the old buffer to at least the same length as the new data.